plugins/frontend/backstage-plugin-argo-cd/src/components/useAppDetails.ts

```diff
@@ -36,57 +35,13 @@ export const useAppDetails = ({
   const errorApi = useApi(errorApiRef);
   const configApi = useApi(configApiRef);
 
-  const getRevisionHistoryDetails = async (
-    appDetails: ArgoCDAppDetails,
-    app: string,
-    instanceName?: string | undefined,
-  ) => {
-    const promises: Promise<void>[] | undefined =
-      appDetails.status?.history?.map(async (historyRecord: any) => {
-        let revisionID = '';
-        if (historyRecord.revision.hasOwnProperty('revisionID')) {
-          revisionID = historyRecord.revision.revisionID;
-        } else {
-          revisionID = historyRecord.revision;
-        }
-        if (historyRecord.source?.chart !== undefined) {
-          historyRecord.revision = { revisionID: revisionID };
-          return;
-        }
-        const revisionDetails = await api.getRevisionDetails({
-          url,
-          app,
-          appNamespace: appDetails.metadata.namespace,
-          revisionID,
-          instanceName,
-        });
-        historyRecord.revision = { revisionID: revisionID, ...revisionDetails };
-      });
-    await Promise.all(promises as readonly unknown[] | []);
-    return appDetails;
-  };
-
   const { loading, value, error, retry } = useAsyncRetry(async () => {
     const argoSearchMethod: boolean = Boolean(
       configApi.getOptionalConfigArray('argocd.appLocatorMethods')?.length,
     );
     try {
       if (!argoSearchMethod && appName) {
-        const appDetails = await api.getAppDetails({
-          url,
-          appName,
-          appNamespace,
-        });
-        if (
-          appDetails &&
-          appDetails?.status &&
-          appDetails?.status?.history &&
-          appDetails?.status?.history.length > 0
-        ) {
-          return getRevisionHistoryDetails(appDetails, appName);
-        }
-
-        return appDetails;
+        return await api.getAppDetails({ url, appName, appNamespace });
```

Comment:
Don't [return await](https://eslint.org/docs/latest/rules/no-return-await). It's an anti-pattern.

Reply:
Looks fine according to the link:
```
This rule was deprecated in ESLint v8.46.0 with no replacement. The original intent of this rule no longer applies due to the fact JavaScript now handles native Promises differently. It can now be slower to remove await rather than keeping it. More technical information can be found in [this V8 blog entry](https://v8.dev/blog/fast-async).
```